src/domain/usecases/DownloadTemplateUseCase.ts

```diff
@@ -158,7 +158,7 @@ export class DownloadTemplateUseCase implements UseCase {
             type,
             id,
             populate,
-            dataPackage,
+            dataPackage: dataPackage && templateFromDataPackage(dataPackage),
```

Comment:
not important, but this is a pattern (typical in Python) that we tend not to use (as && conveys "logical check", but we are doing another thing). I'd use the normal ternary operator.